### PR TITLE
Add explanation for XAI & minor doc fixes

### PR DESCRIPTION
**Pull request by GalyaZalesskaya**
### Summary

- Added explanation for XAI (explainable AI)
- Updated tutorials for XAI
- Added accuracy characteristics for object detector models on VOC dataset (CVS-103636)
- Updates Quick Start and ReadME with XAI mention (requested by Mark)
- Updated structure representation of tutorials not to expose inner headings in documents (the structure became more clean and clear comparing to [current version](https://openvinotoolkit.github.io/training_extensions/latest/guide/tutorials/index.html))
image
- Minor fixes with warnings and broken links

### How to test

cd docs
make html

open build/index.html

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md)
- [x] I have updated the [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: docs/source/guide/explanation/additional_features/xai.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
+Explainable AI (XAI)
+====================
+
+**Explainable AI (XAI)** is a field of research that aims to make machine learning models more transparent and interpretable to humans.
+The goal is to help users understand how and why AI systems make decisions and provide insight into their inner workings. It allows us to detect, analyze, and prevent common mistakes like the lack of data diversity for certain objects.
```
Comment (negvet) [inline docs/source/guide/explanation/additional_features/xai.rst]:
```suggestion
The goal is to help users understand how and why AI systems make decisions and provide insight into their inner workings. It allows us to detect, analyze, and prevent common mistakes such as when the model uses irrelevant features to make a prediction.
```

File: docs/source/guide/explanation/additional_features/xai.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
+Explainable AI (XAI)
+====================
+
+**Explainable AI (XAI)** is a field of research that aims to make machine learning models more transparent and interpretable to humans.
+The goal is to help users understand how and why AI systems make decisions and provide insight into their inner workings. It allows us to detect, analyze, and prevent common mistakes like the lack of data diversity for certain objects. 
+XAI can help to build trust in AI, make sure that the model is safe for development and increase its adoption in various domains.
+
+Most XAI tools generate **saliency maps** as a part of the process. It is a visual representation, suitable for human comprehension, that highlights the most important parts of the image that the network has focused on the most.
```
Comment (negvet) [inline docs/source/guide/explanation/additional_features/xai.rst]:
```suggestion
Most XAI methods generate **saliency maps** as a result. Saliency map is a visual representation, suitable for human comprehension, that highlights the most important parts of the image from the model point of view. 
```

File: docs/source/guide/explanation/additional_features/xai.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
+Explainable AI (XAI)
+====================
+
+**Explainable AI (XAI)** is a field of research that aims to make machine learning models more transparent and interpretable to humans.
+The goal is to help users understand how and why AI systems make decisions and provide insight into their inner workings. It allows us to detect, analyze, and prevent common mistakes like the lack of data diversity for certain objects. 
+XAI can help to build trust in AI, make sure that the model is safe for development and increase its adoption in various domains.
+
+Most XAI tools generate **saliency maps** as a part of the process. It is a visual representation, suitable for human comprehension, that highlights the most important parts of the image that the network has focused on the most. 
+It looks like a heatmap, where warm-colored areas represent the areas with main focuses.
```
Comment (negvet) [inline docs/source/guide/explanation/additional_features/xai.rst]:
```suggestion
It looks like a heatmap, where warm-colored areas represent the areas with main focus.
```

File: docs/source/guide/explanation/additional_features/xai.rst [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+.. image:: ../../../../utils/images/xai_example.jpg
+  :width: 600
+  :alt: this image shows the result of XAI algorithm
+
+
+We can generate saliency maps for a certain model that was trained in OpenVINO™ Training Extensions, using ``otx explain`` command line. Learn more about its usage in  :doc:`../../tutorials/base/explain` tutorial.
+
+*************************
+Classification algorithms
```
Comment (negvet) [inline docs/source/guide/explanation/additional_features/xai.rst]:
```suggestion
XAI algorithms for classification
```

File: docs/source/guide/explanation/additional_features/xai.rst [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+| Single-shot                               | Yes            | Yes            |  No (re-infer neck + head H*W times, where HxW – feature map size)      |                                                          
++-------------------------------------------+----------------+----------------+-------------------------------------------------------------------------+
+| Class discrimination                      | No             | No             | Yes                                                                     |
++-------------------------------------------+----------------+----------------+-------------------------------------------------------------------------+
+| Execution speed                           | Fast           | Fast           | Medium                                                                  |  
++-------------------------------------------+----------------+----------------+-------------------------------------------------------------------------+
+
+
+*************************
+Detection algorithms
```
Comment (negvet) [inline docs/source/guide/explanation/additional_features/xai.rst]:
```suggestion
XAI algorithms for detection
```

File: docs/source/guide/explanation/additional_features/xai.rst [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+*************************
+
+To generate a saliency map for the detection task, we use the **DetClassProbabilityMap** algorithm.
+It's the naive approach for detection that takes the raw classification head output and uses class probability maps to calculate regions of interest for each class. So, it creates different salience maps for each class.
+For now, this algorithm is implemented for single-stage detectors only.​
```
Comment (negvet) [inline docs/source/guide/explanation/additional_features/xai.rst]:
```suggestion
This algorithm is implemented for single-stage detectors only.​ For two-stage detectors **Activation Map​** (see above) is used.
```

File: docs/source/guide/explanation/additional_features/xai.rst [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+.. image:: ../../../../utils/images/xai_det.jpg
+  :width: 600
+  :alt: this image shows the detailed description of XAI detection algorithm
+
+
+The main limitation of this method is that, due to training loss design, activation values drift towards the center of the object. It limits the getting of clear explanations in the near-edge image areas.​
```
Comment (negvet) [inline docs/source/guide/explanation/additional_features/xai.rst]:
```suggestion
The main limitation of this method is that, due to training loss design of most single-stage detectors, activation values drift towards the center of the object while propagating through the network. This prevents from getting clear explanation in the input image space using intermediate activations.
```

File: docs/source/guide/explanation/additional_features/xai.rst [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
++-------------------------------------------+-------------------------------------------------------------------------+
```
Comment (negvet) [inline docs/source/guide/explanation/additional_features/xai.rst]: It is good to add to this table Activation Map algorithm (which is used for two-stage detectors).

**Review by negvet — CHANGES_REQUESTED** [review]
Thank you! It would be good to center images, although not a must

File: docs/source/guide/tutorials/base/explain.rst [hunk 9]
```diff
@@ -26,9 +26,25 @@ at the path specified by ``--save-explanation-to``.
...
 
-3. As a result we will get a folder with a pair of generated 
+3. To specify the algorithm of saliency map creation for classification, 
+we can define the ``--explain-algorithm`` parameter.
+
+- ``activationmap`` - for activation map classification algorithm 
+- ``eigencam`` -  for Eigen-Cam classification algorithm
+- ``classwisesaliencymap`` -  for Recipro-CAM classification algorithm, this is a default method
+
+For detection task, the ``classwisesaliencymap`` is only supported, so we don't need to specify it.
```
Comment (negvet) [inline docs/source/guide/tutorials/base/explain.rst]: `activationmap` is used for two-stage detectors

Comment (negvet) [reply docs/source/guide/tutorials/base/explain.rst]: https://github.com/openvinotoolkit/training_extensions/blob/11a88370f49dbff5c9672867bfb53b1e469f2a22/otx/mpa/det/inferrer.py#L179-L180

**Review by eunwoosh — DISMISSED** [review]

**Comment by GalyaZalesskaya** [comment]
@nevget Thank you for your review and clarification. All done and images were aligned to center

**Review by negvet — DISMISSED** [review]

**Comment by GalyaZalesskaya** [comment]
@eunwoosh  I'm sorry to bother you, but can you kindly review this again?

File: docs/source/guide/explanation/algorithms/object_detection/object_detection.rst [hunk 10]
```diff
@@ -133,7 +137,7 @@ Overall, OpenVINO™ Training Extensions utilizes powerful techniques for improv
 
 Please, refer to the :doc:`tutorial <../../../tutorials/advanced/semi_sl>` how to train semi supervised learning.
 
-In the table below the mAP on toy data sample from `COCO <https://cocodataset.org/#home>`_ dataset using our pipeline is presented.
+In the table below the mAP on toy data sample from `COCO <https://cocodataset.org/#home>`__ dataset using our pipeline is presented.
```
Comment (kprokofi) [inline docs/source/guide/explanation/algorithms/object_detection/object_detection.rst:141]: is this double underscore expected?

File: docs/source/guide/explanation/algorithms/object_detection/object_detection.rst [hunk 11]
```diff
@@ -95,20 +95,24 @@ To see which public backbones are available for the task, the following command
...
-.. +-----------+------------+-----------+-----------+
-.. | Model name| COCO       | PASCAL VOC| MinneApple|
-.. +===========+============+===========+===========+
-.. | YOLOX     | N/A        | N/A       | 24.5      |
-.. +-----------+------------+-----------+-----------+
-.. | SSD       | N/A        | N/A       | 31.2      |
-.. +-----------+------------+-----------+-----------+
-.. | ATSS      | N/A        | N/A       | 42.5      |
-.. +-----------+------------+-----------+-----------+
+For `COCO <https://cocodataset.org/#home>`__ dataset the accuracy of pretrained weights was shown. That means that weights are undertrained for COCO dataset and don't achieve the best result.
```
Comment (kprokofi) [inline docs/source/guide/explanation/algorithms/object_detection/object_detection.rst]:
```suggestion
For `COCO <https://cocodataset.org/#home>`__ dataset the accuracy of pretrained weights is shown. That means that weights are undertrained for COCO dataset and don't achieve the best result. 
```
I think we should align with present tense here

**Review by kprokofi — COMMENTED** [review]
Thank you for your work!
Let's add more datasets for detection as we discussed

File: docs/source/guide/explanation/algorithms/object_detection/object_detection.rst (hunk 10 above)
Comment (GalyaZalesskaya) [reply docs/source/guide/explanation/algorithms/object_detection/object_detection.rst:141]: Yes, in other way I get a warning that I make 2 links to one COCO source (it's cited in Self-SL as well)

File: docs/source/guide/explanation/algorithms/object_detection/object_detection.rst (hunk 11 above)
Comment (GalyaZalesskaya) [reply docs/source/guide/explanation/algorithms/object_detection/object_detection.rst]: Done

**Review by kprokofi — DISMISSED** [review]
LGTM

**Review by negvet — DISMISSED** [review]

**Review by eunwoosh — DISMISSED** [review]

**Review by goodsong81 — CHANGES_REQUESTED** [review]
What do you think of changing the target branch to release/1.1.0?
Or, would you replicate this PR?

**Comment by GalyaZalesskaya** [comment]
@goodsong81 I updated the target branch, thank you

**Review by goodsong81 — APPROVED** [review]

**Pull request merged**